src/simulators/statevector/qubitvector_thrust.hpp

```diff
@@ -369,69 +1106,64 @@ class QubitVectorThrust {
   //-----------------------------------------------------------------------
   // Statevector update with Lambda function
   //-----------------------------------------------------------------------
-  	template <typename UnaryFunction>
-	void apply_function(UnaryFunction func,const reg_t &qubits);
+  template <typename Function>
+  double apply_function(Function func,const reg_t &qubits) const;
+
+  void set_matrix(const cvector_t<double>& mat) const;
+  void set_params(const reg_t& prm) const;
+
+  void allocate_buffers(int qubit);
 
-  	template <typename UnaryFunction>
-	double apply_sum_function(UnaryFunction func,const reg_t &qubits) const;
+  int m_iDev;
+  int m_nDev;
+  int m_nDevParallel;
+  int m_iPlaceHost;
+  int m_nPlaces;
 
-	void allocate_buffers(int qubit);
+  mutable std::vector<QubitVectorChunkContainer<data_t>> m_Chunks;
 
-	int m_iDev;
-	int m_nDev;
-	int m_nDevParallel;
-	int m_useATS;
-	int m_useDevMem;
+  int m_maxChunkBits;           //bits per chunk
+  uint_t m_globalSize;          //number of total states
+  uint_t m_localSize;           //number of states in this process
+  uint_t m_globalIndex;         //starting state ID for this process
+  int m_maxNumBuffers;          //max number of buffer chunks
 
-	thrust::complex<double>* m_pMatDev;		//matrix buffer for device
-	uint_t* m_pUintBuf;						//buffer to store parameters
-	thrust::complex<data_t>** m_ppBuffer;	//pointer to buffers
-	int m_matBits;							//max number of fusion bits
-	uint_t m_matSize;
+  mutable uint_t m_refPosition;         //position for reference (if >= data_size_ data_ is empty)
 
-#ifdef DEBUG
-	mutable uint_t m_gateCounts[QS_NUM_GATES];
-	mutable double m_gateTime[QS_NUM_GATES];
-	mutable double m_gateStartTime[QS_NUM_GATES];
 
-	void TimeStart(int i) const;
-	void TimeEnd(int i) const;
-	void TimeReset(void);
-	void TimePrint(void);
+  int FindPlace(uint_t chunkID,int chunkBits) const;
+  int GlobalToLocal(uint_t& lcid,uint_t& lid,uint_t gid,int chunkBits) const;
+  uint_t GetBaseChunkID(const uint_t gid,const reg_t& qubits,const int chunkBits) const;
 
-	//for debugging
-	mutable FILE* debug_fp;
-	mutable uint_t debug_count;
+  void UpdateReferencedValue(void) const;
 
-	void DebugMsg(const char* str,const reg_t &qubits) const;
-	void DebugMsg(const char* str,const int qubit) const;
-	void DebugMsg(const char* str) const;
-	void DebugMsg(const char* str,const std::complex<double> c) const;
-	void DebugMsg(const char* str,const double d) const;
-	void DebugMsg(const char* str,const std::vector<double>& v) const;
-	void DebugDump(void);
+#ifdef AER_TIMING
```

Comment:
Timing and debugging should really use the result metadata for output, or the C++ logging library so it works through python too. This can be a TODO for the future.

Can you write a short comment here along lines of 
// Print timing for profiling if environment variable <.> is set.